tests/utils/TestCase/ImagesTestCase.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace PerformanceLab\Tests\TestCase;
+
+use PerformanceLab\Tests\Constraint\ImageHasSizeSource;
+use PerformanceLab\Tests\Constraint\ImageHasSource;
+use WP_UnitTestCase;
+
+/**
+ * A test case for image attachments.
+ *
+ * @method void assertImageHasSource( $mime_type, $attachment_id, $message ) Asserts that the image has the appropriate source.
+ * @method void assertImageHasSizeSource( $mime_type, $size, $attachment_id, $message ) Asserts that the image has the appropriate source for the subsize.
+ * @method void assertImageNotHasSource( $mime_type, $attachment_id, $message ) Asserts that the image doesn't have the appropriate source.
+ * @method void assertImageNotHasSizeSource( $mime_type, $size, $attachment_id, $message ) Asserts that the image doesn't have the appropriate source for the subsize.
+ */
+abstract class ImagesTestCase extends WP_UnitTestCase {
+
+	/**
+	 * Asserts that an image has a source with the specific mime type.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $mime_type The mime type of the source.
+	 * @param int    $attachment_id The attachment ID.
+	 * @param string $message An optional message to show on failure.
+	 */
+	public static function assertImageHasSource( $mime_type, $attachment_id, $message = '' ) {
```

Comment:
I would update the parameters as the mime type depends on the `$attachment_id` it makes more sense to set first the `attachment_id`

```suggestion
	public static function assertImageHasSource( $attachment_id, $expected_mime_type, $message = '' ) {
```

Reply:
The current order is used to mimic how the `ArrayHasKey` constraint work. The first param/params are needly and then goes the target that we want to check.

Reply:
Right, however by using this abstraction the second argument is no longer an array and if you are unfamiliar with the abstraction itself is difficult to know which is the base of the abstraction like if I want to use the assertion I would expect the first parameter to be the attachment ID similarly as how WP functions work due the expected_mime_type and additional parameters are based out of the attachment ID.

The logic comes from importance order:

- Attachment ID -> This is required and would always be present 
- Size -> This is also present as is a WP API the only difference is the size name in some cases might not be defined
- Mime Type -> This is a newly introduced value so it might not be present in most cases. 

Based on that order it makes more sense to pass the parameters in that order instead of the other way around.

Reply:
I don't have a strong opinion on either, but slightly leaning towards @mitogh's point here, since that order is what's more common in WordPress image functions.

The argument around `assertArrayHasKey` makes sense, but here we're passing the attachment ID, and the assertion is not looking for the size "in" the attachment ID. It's looking for it in the metadata array. If the assertion was `assertMetadataHasSource`, I think the `assertArrayHasKey` comparison would make more sense, but because we're passing an attachment ID here, IMO sticking to the order WordPress developers are used to makes more sense.

Reply:
Ok, sounds good. Updated.